src/shogun/base/AnyParameter.h

```diff
@@ -31,14 +32,17 @@ namespace shogun
 	};
 
 	/** parameter properties */
-	enum class ParameterProperties
+	enum class ParameterProperties: std::uint32_t
 	{
 		NONE = 0,
 		HYPER = 1u << 0,
 		GRADIENT = 1u << 1,
 		MODEL = 1u << 2,
 		AUTO = 1u << 10,
-		READONLY = 1u << 11
+		READONLY = 1u << 11,
+		Last,
+		ALL = std::numeric_limits<std::uint32_t>::max(),
```

Comment:
Did this work well then? Is there anyway to infer `uint32_t` from the inheritance btw?

Reply:
should probably add a unittest for this?